core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -203,11 +204,24 @@ object Executor {
         isTopLevelField: Boolean
       ) = {
 
-        def collectAllQueries() =
+        def collectAllQueries() = {
+          def combineQueryResults(results: List[ResponseValue]) = {
+            val builder = ListBuffer.empty[(String, ResponseValue)]
+            var resps   = results
+            var names   = steps
+            while (resps ne Nil) {
```

Comment:
Would the same trick be useful here?

Reply:
For consistency, I used the same pattern across all the methods in the Executor where we manually loop over the elements of a List